memory_reference/include/mem_ref_detail.hpp

```diff
@@ -21,6 +21,17 @@ struct Mem_Ref_Detail<CUDAGPU>{
 template <class T>
 using Mem_Ref = typename Mem_Ref_Detail<T>::type;
 
+std::vector<std::tuple<std::string, bool>> get_available_device()
+{
+  std::vector<std::tuple<std::string, bool>> device_list;
+  device_list.push_back({"CPU", true});
+
+  #ifdef ENABLED_CUDA
+  device_list.push_back({"GPU-CUDA", true});
+  #endif
```

Comment:
This needs also an else branch, which adds a `{"GPU-CUDA", false}`

Reply:
By the way, that is also the reason, why this line of code will never triggered: https://github.com/ComputationalRadiationPhysics/student_project_python_bindings/blob/ecb414230006f3bfd32f68d449176fc79e47bd78/memory_reference/src/main.py#L16